esp-hal/src/dma/mod.rs

```diff
@@ -1783,14 +1775,14 @@ where
         peri: DmaPeripheral,
         chain: &DescriptorChain,
     ) -> Result<(), DmaError> {
-        if self.burst_mode
-            && chain
-                .descriptors
-                .iter()
-                .any(|d| d.len() % 4 != 0 || d.buffer as u32 % 4 != 0)
-        {
-            return Err(DmaError::InvalidAlignment);
-        }
+        // if self.burst_mode
```

Comment:
Is this deliberately commented out?

Reply:
yes, but I plan to resolve this in this PR before we I'd call it ready.

Reply:
Okay, so, we don't have enough information to restore this - for I2S currently burst mode will be disabled. We can restore burst mode by rewriting the driver to use the buffer preparation API.

Reply:
Oh somehow missed me mentioned here :) t.b.h. I don't know if it is that important for real-world users - I could imagine it's not important since audio with a sample frequency of 192kHz (which is not that much) is the maximum used - more common are 44.1 and 48kHz. Maybe when abusing the peripheral to do something other than audio

Reply:
Later PRs restore burst mode, so it was only a concern if we were going to merge it in the last release.